trace_replay/trace_replay.cc

```diff
@@ -575,13 +578,17 @@ Status Replayer::Replay() {
         return Status::Corruption("Invalid Column Family ID.");
       }
 
+      Iterator* single_iter = nullptr;
       if (iter_payload.cf_id == 0) {
         single_iter = db_->NewIterator(roptions);
       } else {
         single_iter = db_->NewIterator(roptions, cf_map_[iter_payload.cf_id]);
       }
       single_iter->Seek(iter_payload.iter_key);
-      ops++;
+      s = single_iter->status();
+      if (!s.ok()) {
+        break;
+      }
       delete single_iter;
```

Comment:
Doesn't the delete need to come before the break?

Reply:
It needs to also be present in the body of the if expression. I have now fixed this